kie-api/src/main/java/org/kie/api/internal/utils/ServiceDiscoveryImpl.java

```diff
@@ -140,14 +141,16 @@ private void processKieService(ClassLoader classLoader, String key, String value
         }
     }
 
-    private Map<String, Object> buildMap() {
-        Map<String, Object> servicesMap = new HashMap<>();
-        for (Map.Entry<String, Object> serviceEntry : services.entrySet()) {
+    private Map<String, List<Object>> buildMap() {
```

Comment:
@mariofusco 
All that PR resolves to the population of that map.
I think we should put the required information in one single place, easy to find and modify.
Polluting the code with "priority information" scattered all over the codebase just add unneeded complexity and decrease the overall maintainability, without a real benefit.